caffe2/core/operator_c10wrapper.h

```diff
@@ -146,31 +133,86 @@ class C10OperatorWrapper final : public Operator<Context> {
     std::vector<at::Tensor> result;
     result.reserve(InputSize());
     for (size_t i = 0; i < InputSize(); ++i) {
-      result.push_back(at::Tensor(c10::C10Tensor(Input(i))));
+      result.emplace_back(Input(i));
+    }
+    return result;
+  }
+
+  std::vector<at::Tensor> preallocated_outputs_() {
+    std::vector<at::Tensor> result;
+    result.reserve(OutputSize());
+    for (size_t i = 0; i < OutputSize(); ++i) {
+      result.emplace_back(OperatorBase::OutputTensorOrUndefined(i));
     }
     return result;
   }
 
+  IValue get_nontensor_argument_(const c10::Argument& argument) {
```

Comment:
This method looks very copy pastey. OK for now but worth seeing if it can be simplified.